pkg/ephemeral/io/tuple_streamer.go

```diff
@@ -273,15 +272,11 @@ func (ts *CastorTupleStreamer) bufferData(terminateCh chan struct{}, streamerErr
 func (ts *CastorTupleStreamer) getTupleData() ([]byte, error) {
 	requestID := uuid.NewMD5(ts.baseRequestID, []byte(strconv.Itoa(ts.requestCycle)))
 	ts.requestCycle++
-	tupleList, err := ts.castorClient.GetTuples(ts.stockSize, ts.tupleType, requestID)
+	tupleData, err := ts.castorClient.GetTuples(ts.stockSize, ts.tupleType, requestID)
 	if err != nil {
 		return nil, err
 	}
 	ts.logger.Debugw("Fetched new tuples from Castor", "RequestID", requestID)
-	tupleData, err := ts.tupleListToByteArray(tupleList)
```

Comment:
I think we should at least check whether received data fits to the requested tuple parameters (length of array equals number of requested tuples x length of requested tuple type, or sth like that)